src/hooks.ts

```diff
@@ -36,51 +37,58 @@ export const createAPIHooks = <Endpoints extends RoughEndpoints>({
     useAPIQuery: (route, payload, options) => {
       const client = useClient();
       const queryKey: QueryKey = [createQueryKey(name, route, payload)];
-      return useQuery(
+
+      const query = useQuery({
+        ...options,
         queryKey,
-        () =>
+        queryFn: () =>
           client
             .request(route, payload, options?.axios)
             .then((res) => res.data),
-        options,
-      );
+      });
+
+      return query;
     },
+
     useInfiniteAPIQuery: (route, initPayload, options) => {
       const client = useClient();
       const queryKey: QueryKey = [
         INFINITE_QUERY_KEY,
         createQueryKey(name, route, initPayload),
       ];
-      const query = useInfiniteQuery(
+
+      const query = useInfiniteQuery({
+        ...options,
         queryKey,
-        ({ pageParam }) => {
+        initialPageParam: options.initialPageParam,
```

Comment:
nit: any particular reason we need to specify this here rather than getting it from the spread on line 61?